### ISSUE TITLE
Q: How are each of these CFG hacks different from one another?

### ISSUE THREAD

**Issue by TomLucidor** [v0xie/sd-webui-incantations#63]
Are there explainers on how each of these methods vary in quality?
- Smoothed Energy Guidance
- Semantic CFG
- Perturbed Attention Guidance
- CFG Interval / CFG Scheduler ("Clamp")

Bonus: what about these other new methods?
- Dynamic Thresholding ("X mimic Y clamp Z%") https://github.com/mcmonkeyprojects/sd-dynamic-thresholding
- CFG++ or "manifold-constrained" https://github.com/CFGpp-diffusion/CFGpp
- Skimmed CFG https://github.com/Extraltodeus/Skimmed_CFG
- CFG Rescale https://github.com/Seshelle/CFG_Rescale_webui https://github.com/MythicalChu/CFGRescale_For_Forge
- APG Scaling https://github.com/logtd/ComfyUI-APGScaling

**Comment by 6DammK9** [comment]
Currently moving from "FreeU + DynCFG + PAG" to "CFG++ + PAG + SEG". You can see my findings in [my repo](https://github.com/6DammK9/nai-anime-pure-negative-prompt/blob/main/ch01/cfgpp_pag_seg.md).
Notice that I have agressive parameters so it may not match to "general usage".

Currently [CFG++](https://github.com/AUTOMATIC1111/stable-diffusion-webui/pull/16035) is a new topic to explore, there is an "optimimal STEPs" to search, even it should be a common value coincidentally.

~~Trust me, [I have a model very capable in this topic](https://civitai.com/models/309514/astolfomix-xl), unconditional image generation is a deep topic, and I believe in merging.~~

**Comment by TomLucidor** [comment]
Pretty lit @6DammK9 (wishing for an ELI5-ish article but oh well ChatGPT time)... Wait I think I missed these ones 
- Characteristic Guidance https://github.com/scraed/CharacteristicGuidanceWebUI
- Particle Guidance https://github.com/gcorso/particle-guidance
- "Attend and Excite" (not even sure if this is relevant) https://github.com/yuval-alaluf/Attend-and-Excite

**Comment by 6DammK9** [comment]
Characteristic Guidance = CHG. Tested, too long for inference. Roll 10 seeds instead.
Next 2 has no "UI" integration so no comment.
